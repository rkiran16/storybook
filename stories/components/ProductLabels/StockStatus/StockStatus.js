import { html } from "lit-html";

export function StockStatus({ title, variation, count }) {
  return html`
    <div>
      <span
        class="sub-title-3 rounded-pill p-1 ${count < 10
          ? "px-2"
          : ""} bg-${variation} text-white "
        >${count}</span
      >
      <span class=${`sub-title-3 text-${variation}`}>${title}</span>
    </div>
  `;
}
