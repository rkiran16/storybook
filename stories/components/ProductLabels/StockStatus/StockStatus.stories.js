import { StockStatus } from "./StockStatus";

export default {
  title: "Components/ProductLabels/StockStatus",
  parameters: {
    layout: "centered",
    zeplinLink:
      "https://app.zeplin.io/project/62902a463701ffb0e532df2b/screen/63053ee9c9b39414a05fab9b",
  },
  argTypes: {
    variation: {
      name: "Color Variations",
      control: { type: "select" },
      description: "Assign Color Variations",
      options: ["success", "danger", "warning", "info"],
      table: {
        type: { summary: "select" },
        defaultValue: { summary: "info" },
      },
    },
  },
};

const StockStatusTemplate = (args) => StockStatus(args);

export const InStock = StockStatusTemplate.bind({});
InStock.args = {
  title: "IN STOCK",
  variation: "success",
  count: 378,
};
export const LowStock = StockStatusTemplate.bind({});
LowStock.args = {
  title: "LOW STOCK",
  variation: "warning",
  count: 18,
};
export const OutOfStock = StockStatusTemplate.bind({});
OutOfStock.args = {
  title: "OUT OF STOCK",
  variation: "danger",
  count: 0,
};

export const AlertBadge = StockStatusTemplate.bind({});
AlertBadge.args = {
  title: "",
  variation: "info",
  count: 7,
};
