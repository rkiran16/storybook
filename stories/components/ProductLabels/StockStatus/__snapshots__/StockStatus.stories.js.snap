// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Components/ProductLabels/StockStatus Alert Badge 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
    <div>
      <span
        class=\\"sub-title-3 rounded-pill p-1 ",
    " bg-",
    " text-white \\"
        >",
    "</span
      >
      <span class=",
    ">",
    "</span>
    </div>
  ",
  ],
  "values": Array [
    "px-2",
    "info",
    7,
    "sub-title-3 text-info",
    "",
  ],
}
`;

exports[`Storyshots Components/ProductLabels/StockStatus In Stock 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
    <div>
      <span
        class=\\"sub-title-3 rounded-pill p-1 ",
    " bg-",
    " text-white \\"
        >",
    "</span
      >
      <span class=",
    ">",
    "</span>
    </div>
  ",
  ],
  "values": Array [
    "",
    "success",
    378,
    "sub-title-3 text-success",
    "IN STOCK",
  ],
}
`;

exports[`Storyshots Components/ProductLabels/StockStatus Low Stock 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
    <div>
      <span
        class=\\"sub-title-3 rounded-pill p-1 ",
    " bg-",
    " text-white \\"
        >",
    "</span
      >
      <span class=",
    ">",
    "</span>
    </div>
  ",
  ],
  "values": Array [
    "",
    "warning",
    18,
    "sub-title-3 text-warning",
    "LOW STOCK",
  ],
}
`;

exports[`Storyshots Components/ProductLabels/StockStatus Out Of Stock 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
    <div>
      <span
        class=\\"sub-title-3 rounded-pill p-1 ",
    " bg-",
    " text-white \\"
        >",
    "</span
      >
      <span class=",
    ">",
    "</span>
    </div>
  ",
  ],
  "values": Array [
    "px-2",
    "danger",
    0,
    "sub-title-3 text-danger",
    "OUT OF STOCK",
  ],
}
`;
